src/System.Private.ServiceModel/src/System/ServiceModel/Channels/LayeredChannel.cs

```diff
@@ -57,12 +58,12 @@ protected override void OnClose(TimeSpan timeout)
 
         protected override IAsyncResult OnBeginClose(TimeSpan timeout, AsyncCallback callback, object state)
         {
-            return _innerChannel.BeginClose(timeout, callback, state);
+            throw ExceptionHelper.PlatformNotSupported();
```

Comment:
Sounds like a plan. Wondering if a Contract.Requires would help with static analysis as well? 

---
In reply to: [78079821](https://github.com/dotnet/wcf/pull/1515#discussion_r78079821) [](ancestors = 78079821)